ModulePresentationsForCAP/gap/ModulePresentationsForCAP.gi

```diff
@@ -2146,3 +2172,336 @@ end );
     end, 1000 );
 
 end );
+
+##
+## Random methods
+##
+
+##
+## Interpretation of n:
+## The number of rows and colms of the matrix of the object is less or equal to n.
+InstallGlobalFunction( ADD_RANDOM_OBJECT,
+
+  function( category, left_or_right )
+
+    AddRandomObject( category,
+
+      function( C, n )
+        local nr_rows, nr_cols, mat, homalg_ring;
+
+        homalg_ring := category!.ring_for_representation_category;
+        
+        if n < 0 then
+          
+          return fail;
+
+        fi;
+
+        if n = 0 then
+          
+          return ZeroObject( category );
+
+        fi;
+
+        nr_rows := Random( [ 1 .. n ] );
+
+        nr_cols := Random( [ 1 .. n ] );
+
+        mat := homalg_ring!.random_matrix_func( nr_rows, nr_cols );
+
+        if left_or_right = "left" then
+          
+          return AsLeftPresentation( mat );
+
+        else
+
+          return AsRightPresentation( mat );
+
+        fi;
+
+      end );
+
+end );
+
+##
+## Interpretation of n:
+## The number of relations of the range of the created random morphism is n.
+InstallGlobalFunction( ADD_RANDOM_MORPHISM_WITH_FIXED_SOURCE_LEFT,
+    function( category )
+      local homalg_ring;
+
+      homalg_ring := category!.ring_for_representation_category;
+
+      AddRandomMorphismWithFixedSource( category,
+        function( M, n )
+          local m, y, syz, x, u, U, e;
+
+          if n < 0 then
+            
+            return fail;
+
+          fi;
+
+          m := UnderlyingMatrix( M );
+
+          y := homalg_ring!.random_matrix_func( NrRows( m ), n );
+
+          syz := SyzygiesOfColumns( UnionOfColumns( m, y ) );
+
```

Comment:
If you want to, you could improve the algorithm here by computing an arbitrary random matrix that you precompose with `SyzygiesOfColumns`, instead chopping of columns and multiplying them with `e`.

Reply:
Thank you for the suggestion. It have done that in this commit:

https://github.com/homalg-project/CAP_project/pull/239/commits/dd5cbf7910255ca87872ebd457f8bbf0f14bb5ff